web/src/pages/setting/SettingContent.vue

```diff
@@ -0,0 +1,107 @@
+<script lang="ts" setup>
+export interface LanguageWithIndex {
+  index: number
+}
+export interface LanguageWithIndex extends Language {}
+//LanguageWithIndex interface is merged by the Language type and custome index prop
+
+import { Language, languages } from '@/services/language';
+import { Dialog, DialogPanel, DialogTitle, DialogDescription, TransitionChild, TransitionRoot, TabGroup, TabList, Tab, TabPanels, TabPanel } from '@headlessui/vue'
+import LanguageSelector from '@/components/settings/LangSelector.vue';
+import { ref, onMounted } from 'vue';
+import { useI18n } from 'vue-i18n';
+const { t } = useI18n();
+
+interface props {
+  show: boolean
+}
+
+const props = defineProps<props>()
+
+const categories = ref({
+  'audioLanguage': languages,
+  'subtitleLanguage': languages,
+})
+const audSelected = ref(getLocalAudSelected);
+const subSelected = ref(getLocalSubSelected);
+const curretnCategory = ref(t('audioLanguage')); //inital value of the tab
+
+const emits = defineEmits(["closeDialog"]);
+
+function getLocalAudSelected() {
+  if (localStorage.getItem('audSelected') == null) {
+    localStorage.setItem('audSelected', 'en')
+  }
+  var result: string = localStorage.getItem('audSelected') ?? 'en'
+  return result;
+}
+function getLocalSubSelected() {
+  if (localStorage.getItem('subSelected') == null) {
+    localStorage.setItem('subSelected', 'en')
+  }
+  var result: string = localStorage.getItem('subSelected') ?? 'en'
+  return result;
+}
+function getSelectedOption() {
```

Comment:
This is not reactive.

Use refs and computeds where you need some reactive component and avoid the "function" syntax (use "() => {}" instead).